packages/react-mongo/react-mongo.ts

```diff
@@ -0,0 +1,179 @@
+    params: {
+      name,
+      args
+    }
+  }).current
+
+  if (!EJSON.equals(refs.params, { name, args })) {
+    refs.updateOnReady = false
+    refs.isReady = false
+    refs.params = { name, args }
+  }
+
+  useEffect(() => {
+    const computation = Tracker.nonreactive(() => (
+      Tracker.autorun(() => {
+        const { name, args } = refs.params
+        if (!name) return
+
+        refs.subscription = Meteor.subscribe(name, ...args)
+
+        const isReady = refs.subscription.ready()
+        if (isReady !== refs.isReady) {
+          refs.isReady = isReady
+          if (refs.updateOnReady) {
+            forceUpdate()
+          }
+        }
+      })
+    ))
+
+    return () => {
+      computation.stop()
+      delete refs.subscription
+    }
+  }, [refs.params])
+
+  return [
+    () => {
+      refs.updateOnReady = true
+      return !refs.isReady
+    },
+    refs.subscription
+  ]
+}
+
+const useSubscriptionServer = (name?: string, ...args: any[]): [() => boolean, Meteor.SubscriptionHandle | undefined] => ([
+  () => false,
+  undefined
+])
+
+export const useSubscription = Meteor.isServer
+  ? useSubscriptionServer
+  : useSubscriptionClient
+
+type useFindActions<T> =
+  | { type: 'refresh', data: T[] }
+  | { type: 'addedAt', document: T, atIndex: number }
+  | { type: 'changedAt', document: T, atIndex: number }
+  | { type: 'removedAt', atIndex: number }
+  | { type: 'movedTo', fromIndex: number, toIndex: number }
+
+const useFindReducer = <T>(data: T[], action: useFindActions<T>): T[] => {
+  switch (action.type) {
+    case 'refresh':
+      return action.data
+    case 'addedAt':
+      return [
+        ...data.slice(0, action.atIndex),
+        action.document,
+        ...data.slice(action.atIndex)
+      ]
+    case 'changedAt':
+      return [
+        ...data.slice(0, action.atIndex),
+        action.document,
+        ...data.slice(action.atIndex + 1)
+      ]
+    case 'removedAt':
+      return [
+        ...data.slice(0, action.atIndex),
+        ...data.slice(action.atIndex + 1)
+      ]
+    case 'movedTo':
+      const doc = data[action.fromIndex]
+      const copy = [
+        ...data.slice(0, action.fromIndex),
+        ...data.slice(action.fromIndex + 1)
+      ]
+      copy.splice(action.toIndex, 0, doc)
+      return copy
+  }
+}
+
+const useFindClient = <T = any>(factory: () => Mongo.Cursor<T>, deps: DependencyList) => {
+  let [data, dispatch] = useReducer<Reducer<T[], useFindActions<T>>>(
+    useFindReducer,
+    []
+  )
+
+  const cursor = useMemo(() => {
+    // To avoid creating side effects in render, opt out
+    // of Tracker integration altogether.
+    const c = Tracker.nonreactive(() => factory())
+    if (Meteor.isDevelopment && !(c instanceof Mongo.Cursor)) {
+      console.warn(
+        'Warning: useFindClient requires an instance of Mongo.Cursor. '
+        + 'Make sure you do NOT call fetch() on your cursor.'
+      );
+    }
+    data = c.fetch()
+    return c
```

Comment:
I think this exact block of code should also be executed in `useFindServer`, it doesn't really make sense to not validate it there too. Also rename to `useFind` in the message: "Warning: _useFind_ requires an"  since that's what it is to the user.